packages/zudoku/src/lib/core/ProtectedRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from "react";
+import { matchPath, useLocation } from "react-router";
+import { useAuth } from "../authentication/hook.js";
+import { useZudoku } from "../components/context/ZudokuContext.js";
+import { ZudokuError } from "../util/invariant.js";
+
+export function ProtectedRoute({ children }: { children: React.ReactNode }) {
```

Comment:
Can we name it `RouteGuard` or something? It's a bit confusing in `main.tsx` that it's wrapped around the whole app.